libs/net/dal/Services/FileReferenceService.cs

```diff
@@ -254,5 +251,91 @@ public async Task<IEnumerable<FileReference>> GetFiles(DateTime? updatedBefore =
             return Enumerable.Empty<FileReference>();
         }
     }
-    #endregion
+
+    public async Task<Stream?> DownloadFromS3Async(string s3Key)
+    {
+        if (!_s3Options.IsS3Enabled || await _s3Options.TestNetworkConnectionAsync() == false || _s3Options.S3Client == null)
+        {
+            return null;
+        }
+
+        try
+        {
+            var request = new GetObjectRequest
+            {
+                BucketName = _s3Options.BucketName,
+                Key = s3Key
+            };
+
+            var response = await _s3Options.S3Client.GetObjectAsync(request);
+
+            if (response.HttpStatusCode == System.Net.HttpStatusCode.OK)
+            {
+                var memoryStream = new MemoryStream();
+                await response.ResponseStream.CopyToAsync(memoryStream);
+                memoryStream.Position = 0;
+                return memoryStream;
+            }
+            else
+            {
+                Logger.LogError("Failed to download file from S3: {S3Key}", s3Key);
+                return null;
+            }
+        }
+        catch (Exception)
+        {
+            return null;
+        }
+    }
+
+    public async Task<FileReference> UpdateAsync(FileReference entity)
+    {
+        this.Context.Update(entity);
+        await this.Context.SaveChangesAsync();
+        return entity;
+    }
+
+    public async Task<int> DeleteOldLocalFilesAsync(DateTime beforeDate)
```

Comment:
I still don't understand the purpose of this method related to the business processes.  What is the requirement that requires this function?

Reply:
I will remove it as well. Endpoint was removed.